polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -116,32 +116,31 @@ MLIRScanner::MLIRScanner(MLIRASTConsumer &Glob,
     : Glob(Glob), module(module), builder(module->getContext()),
       loc(builder.getUnknownLoc()), ThisCapture(nullptr), LTInfo(LTInfo) {}
 
-void MLIRScanner::initSupportedConstructors() {
-  // List from SYCLFuncRegistry.cpp Please modify as new constructors are
-  // added to that file.
-  supportedCons.insert("_ZN2cl4sycl2idILi1EEC1Ev");
-  supportedCons.insert("_ZN2cl4sycl2idILi2EEC1Ev");
-  supportedCons.insert("_ZN2cl4sycl2idILi3EEC1Ev");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi1EEC1ILi1EEENSt9enable_ifIXeqT_Li1EEmE4typeE");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi2EEC1ILi2EEENSt9enable_ifIXeqT_Li2EEmE4typeE");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi3EEC1ILi3EEENSt9enable_ifIXeqT_Li3EEmE4typeE");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi1EEC1ILi1EEENSt9enable_ifIXeqT_Li1EEmE4typeEm");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi2EEC1ILi2EEENSt9enable_ifIXeqT_Li2EEmE4typeEm");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi3EEC1ILi3EEENSt9enable_ifIXeqT_Li3EEmE4typeEm");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi1EEC1ILi1EEENSt9enable_ifIXeqT_Li1EEmE4typeEmm");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi2EEC1ILi2EEENSt9enable_ifIXeqT_Li2EEmE4typeEmm");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi3EEC1ILi3EEENSt9enable_ifIXeqT_Li3EEmE4typeEmm");
-  supportedCons.insert("_ZN2cl4sycl6detail5arrayILi1EEC1ILi1EEENSt9enable_"
-                       "ifIXeqT_Li1EEmE4typeE");
+void MLIRScanner::initSupportedFunctions() {
+  // Functions needed for single_task with one dimensional write buffer.
```

Comment:
May be elaborate why this functions are needed in this comment too (copy from the PR description)

Reply:
Added to the declaration.